### PR TITLE
[WIP] refactor: remove EdxRestAPIClient

### PR DESCRIPTION
**Pull request by dyudyunov**
YT: https://youtrack.raccoongang.com/issue/OeX_Depr-58

### CODE REVIEW

File: openedx/core/djangoapps/catalog/management/commands/cache_programs.py [hunk 1]
```diff
@@ -143,11 +146,13 @@ def get_site_program_uuids(self, client, site):  # lint-amnesty, pylint: disable
                 'status': ('active', 'retired'),
                 'uuids_only': 1,
             }
-
-            logger.info(f'Requesting program UUIDs for {site.domain}.')
-            uuids = client.programs.get(**querystring)
+            api_url = urljoin(f"{api_base_url}/", "programs")
+            logger.info('Requesting program UUIDs for %s.', site.domain)
```
Comment (idegtiarov) [inline openedx/core/djangoapps/catalog/management/commands/cache_programs.py:150]: Hm, we're changing f-string in logs to the old style, may we still use the f-string instead?

File: openedx/core/djangoapps/catalog/utils.py [hunk 2]
```diff
@@ -231,11 +241,15 @@ def get_program_types(name=None):
     """
     user, catalog_integration = check_catalog_integration_and_get_user(error_message_field='Program types')
     if user:
-        api = create_catalog_api_client(user)
-        cache_key = f'{catalog_integration.CACHE_KEY}.program_types'
+        cache_key = f"{catalog_integration.CACHE_KEY}.program_types"
```
Comment (idegtiarov) [inline openedx/core/djangoapps/catalog/utils.py:244]: Why do you change the single quotes with double quotes?

File: openedx/core/djangoapps/programs/utils.py [hunk 3]
```diff
@@ -784,7 +790,7 @@ def _collect_one_click_purchase_eligibility_data(self):  # lint-amnesty, pylint:
                     'full_program_price': discount_data['total_incl_tax'],
                     'variant': bundle_variant
                 })
-            except (ConnectionError, SlumberBaseException, Timeout):
+            except (ConnectionError, RequestException, Timeout):
```
Comment (idegtiarov) [inline openedx/core/djangoapps/programs/utils.py:793]: ditto

**Review by idegtiarov — CHANGES_REQUESTED** [review]
Some questions and nits are inline, please take a look. Other LGTM!

File: common/djangoapps/student/models.py [hunk 4]
```diff
@@ -82,15 +74,15 @@
 from lms.djangoapps.courseware.models import (
     CourseDynamicUpgradeDeadlineConfiguration,
     DynamicUpgradeDeadlineConfiguration,
-    OrgDynamicUpgradeDeadlineConfiguration,
+    OrgDynamicUpgradeDeadlineConfiguration
```
[the remark this answers is not in the document]
Comment (dyudyunov) [reply common/djangoapps/student/models.py:77]: I've changed my isort config to use a trailing comma in such cases. Discarded this and all other similar changes

File: openedx/core/djangoapps/catalog/management/commands/cache_programs.py (hunk 1 above)
Comment (dyudyunov) [reply openedx/core/djangoapps/catalog/management/commands/cache_programs.py:150]: Returned the f-string syntax

File: openedx/core/djangoapps/catalog/utils.py (hunk 2 above)
Comment (dyudyunov) [reply openedx/core/djangoapps/catalog/utils.py:244]: reverted )

File: openedx/core/djangoapps/api_admin/views.py [hunk 5]
```diff
@@ -143,90 +162,115 @@ def post(self, request):
...
+        """
         form = CatalogForm(request.POST)
         client = self.get_catalog_api_client(request.user)
         if not form.is_valid():
             return render_to_response(self.template, self.get_context_data(client, username, form), status=400)
 
         attrs = form.cleaned_data
-        catalog = client.catalogs.post(attrs)
+        response = client.post(self.catalogs_api_url, data=attrs)
+        response.raise_for_status()
```
[the remark this answers is not in the document]
Comment (dyudyunov) [reply openedx/core/djangoapps/api_admin/views.py:216]: I think there is no need for additional handling here. 
The view behavior stays the same and any further changes look to be out of the scope of this depr work

Do you think we should improve something here?

File: openedx/core/lib/tests/test_edx_api_utils.py [hunk 6]
```diff
@@ -142,27 +156,28 @@ def test_get_specific_resource(self):
...
         catalog_integration = self.create_catalog_integration()
-        api = create_catalog_api_client(self.user)
+        api = get_catalog_api_client(self.user)
 
         resource_id = 0
-        url = '{api_root}/programs/{resource_id}/'.format(
-            api_root=CatalogIntegration.current().get_internal_api_url().strip('/'),
-            resource_id=resource_id,
-        )
+        url = f"{self.base_api_url}/programs/{resource_id}/"
```
Comment (idegtiarov) [inline openedx/core/lib/tests/test_edx_api_utils.py:180]: ditto

File: openedx/core/lib/tests/test_edx_api_utils.py [hunk 7]
```diff
@@ -232,10 +264,7 @@ def test_cache_utilization(self):
         )
 
         resource_id = 1
-        url = '{api_root}/programs/{resource_id}/'.format(
-            api_root=CatalogIntegration.current().get_internal_api_url().strip('/'),
-            resource_id=resource_id,
-        )
+        url = f"{self.base_api_url}/programs/{resource_id}/"
```
Comment (idegtiarov) [inline openedx/core/lib/tests/test_edx_api_utils.py:267]: ditto

**Review by idegtiarov — CHANGES_REQUESTED** [review]
One minor nit other LGTM! Thanks for the great work!

**Review by idegtiarov — APPROVED** [review]

**Pull request closed without merging**